code/go/0chain.net/miner/protocol_block.go

```diff
@@ -80,10 +80,12 @@ func (mc *Chain) createFeeTxn(b *block.Block) *transaction.Transaction {
 	return feeTxn
 }
 
-func (mc *Chain) getCurrentSelfNonce(minerId datastore.Key, bState util.MerklePatriciaTrieI) (int64, error) {
+func (mc *Chain) getCurrentSelfNonce(round int64, minerId datastore.Key, bState util.MerklePatriciaTrieI) (int64, error) {
 	s, err := mc.GetStateById(bState, minerId)
 	if err != nil {
 		if err != util.ErrValueNotPresent {
+			mc.SyncMissingNodes(round, util.Path(minerId))
```

Comment:
should we check ErrNotFound directly here or do we assume that only 2 types of errors can be return?

Reply:
Yes, should check ErrNotFound directly, could have other errors